frontend/context/StakingProgramProvider.tsx

```diff
@@ -1,73 +1,94 @@
-import { useInterval } from 'usehooks-ts';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { createContext, PropsWithChildren, useCallback } from 'react';
 
-import { CHAIN_CONFIG } from '@/config/chains';
+import { AGENT_CONFIG } from '@/config/agents';
+import { GNOSIS_CHAIN_CONFIG } from '@/config/chains';
+import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
 import { StakingProgramId } from '@/enums/StakingProgram';
+import { useService } from '@/hooks/useService';
 import { useServices } from '@/hooks/useServices';
-import { AutonolasService } from '@/service/Autonolas';
+import { Nullable } from '@/types/Util';
 
-export const INITIAL_DEFAULT_STAKING_PROGRAM_ID = StakingProgramId.Beta;
+export const INITIAL_DEFAULT_STAKING_PROGRAM_ID = StakingProgramId.PearlBeta;
 
 export const StakingProgramContext = createContext<{
   activeStakingProgramId?: StakingProgramId | null;
   defaultStakingProgramId: StakingProgramId;
-  updateActiveStakingProgramId: () => Promise<void>;
-  setDefaultStakingProgramId: (stakingProgramId: StakingProgramId) => void;
 }>({
   activeStakingProgramId: undefined,
   defaultStakingProgramId: INITIAL_DEFAULT_STAKING_PROGRAM_ID,
-  updateActiveStakingProgramId: async () => {},
-  setDefaultStakingProgramId: () => {},
 });
 
-/** Determines the current active staking program, if any */
-export const StakingProgramProvider = ({ children }: PropsWithChildren) => {
-  const { service } = useServices();
+const currentAgent = AGENT_CONFIG.trader; // TODO: replace with dynamic agent selection
+const currentChainId = GNOSIS_CHAIN_CONFIG.chainId; // TODO: replace with dynamic chain selection
 
-  const [activeStakingProgramId, setActiveStakingProgramId] =
-    useState<StakingProgramId | null>();
+/**
+ * hook to get the active staking program id
+ */
+const useGetActiveStakingProgramId = () => {
+  const queryClient = useQueryClient();
 
-  const [defaultStakingProgramId, setDefaultStakingProgramId] = useState(
-    INITIAL_DEFAULT_STAKING_PROGRAM_ID,
-  );
+  const { selectedService, isFetched: isLoaded } = useServices();
+  const serviceConfigId =
+    isLoaded && selectedService ? selectedService?.service_config_id : '';
+  const { service } = useService({ serviceConfigId });
 
-  const updateActiveStakingProgramId = useCallback(async () => {
-    // if no service nft, not staked
-    const serviceId =
-      service?.chain_configs[CHAIN_CONFIG.OPTIMISM.chainId].chain_data?.token;
+  // if no service nft, not staked
+  const serviceId = service?.chain_configs[currentChainId].chain_data?.token;
 
-    if (
-      !service?.chain_configs[CHAIN_CONFIG.OPTIMISM.chainId].chain_data?.token
-    ) {
-      setActiveStakingProgramId(null);
-      return;
-    }
+  const response = useQuery({
+    queryKey: [REACT_QUERY_KEYS.STAKING_PROGRAM_KEY, currentChainId, serviceId],
```

Comment:
shouldn't this be a call `REACT_QUERY_KEYS.STAKING_PROGRAM_KEY(currentChainId, serviceId)`?